src/components/Cluster/ClusterDetail/NodePool.tsx

```diff
@@ -52,41 +53,108 @@ const NameWrapperDiv = styled.div`
   }
 `;
 
-class NodePool extends Component {
-  state = {
+interface INPViewAndEditName {
+  activateEditMode: () => boolean;
+  name: string;
+}
+
+type TNPViewAndEditName = React.Component<INPViewAndEditName>;
+
+interface IStateProps {
+  availableZonesGridTemplateAreas: string | undefined;
+  cluster: object;
```

Comment:
Object is the base type of everything in JavaScript, a `Record` would be more suitable

Reply:
If I don't want to specify the shape because I haven't an interface for it yet, isn't `object` best than `any`?

Reply:
Oops, I was mistaking the `object` type with the `Object` type. The `Object` type is the one that defines all objects, and the `object` type defines all non-primitive types. So both `Record` and `object` alright here, for now

Reply:
hehe, at least I got something right! 😅